In [2]:
import pandas as pd 
import numpy as np
import re
import time
from pandas import DataFrame, Series
from pprint import pprint

In [3]:
df1 = pd.read_excel('./A (15,726).xlsx')
df2 = pd.read_excel('./B (15,826).xlsx')
df3 = pd.read_excel('./C (4,773).xlsx')
df4 = pd.read_excel('./D (14,949).xlsx')
df5 = pd.read_excel('./E (11,087).xlsx')
df6 = pd.read_excel('./F (7,859).xlsx')
df7 = pd.read_excel('./G (7,113).xlsx')
df8 = pd.read_excel('./H (4,949).xlsx')

In [4]:
print('df1의 형태:',df1.shape)
print('df2의 형태:',df2.shape)
print('df3의 형태:',df3.shape)
print('df4의 형태:',df4.shape)
print('df5의 형태:',df5.shape)
print('df6의 형태:',df6.shape)
print('df7의 형태:',df7.shape)
print('df8의 형태:',df8.shape)

df1의 형태: (15726, 19)
df2의 형태: (15826, 20)
df3의 형태: (4773, 19)
df4의 형태: (14949, 19)
df5의 형태: (11087, 19)
df6의 형태: (7859, 19)
df7의 형태: (7113, 19)
df8의 형태: (4949, 19)


In [5]:
# df2 에 Unnamed: 19 라는 컬럼이 한 가지 더 추가되어있는것을 볼 수 있다.
df2

SPEAKER                  SENTENCE      DATAID DOMAINID DOMAIN CATEGORY  \
0          고객          신발은 여기 있는 게 다예요?  박은영_액세서리_5        B  의복의류점       신발   
1          점원           네 성인이나 아동 다 있어요  박은영_액세서리_5        B  의복의류점       신발   
2          점원             발 사이즈 몇 신으세요?  박은영_액세서리_5        B  의복의류점       신발   
3          고객                     230이요  박은영_액세서리_5        B  의복의류점       신발   
4          점원       편하게 신을 수 있는 거 찾으세요?  박은영_액세서리_5        B  의복의류점       신발   
...       ...                       ...         ...      ...    ...      ...   
15821      고객              이 쿠폰에 적립해주세요          생성        B  의복의류점     액세서리   
15822      고객   사은품 받으려면 쿠폰 몇 개 모아야 해요?          생성        B  의복의류점     액세서리   
15823      고객  쿠폰 몇 개 모으면 사은품 받을 수 있어요?          생성        B  의복의류점     액세서리   
15824      고객  쿠폰 몇 장 모아야 사은품 받을 수 있나요?          생성        B  의복의류점     액세서리   
15825      고객   사은품 받으려면 쿠폰 몇 장 있어야 돼요?          생성        B  의복의류점     액세서리   

       SPEAKERID  SENTENCEID         MAIN   SUB QA  QACNCT  \
0              1           1  종류별신발제품문의요청   NaN  Q     NaN   
1              0           2  종류별신발제품문의요청   NaN  A     NaN   
2              0           3  종류별신발제품문의요청   사이즈  Q     NaN   
3              1           4  종류별신발제품문의요청   사이즈  A     NaN   
4              0           5          착화감  제품문의  Q     NaN   
...          ...         ...          ...   ... ..     ...   
15821          1         922      쿠폰시적립요청   NaN  Q     NaN   
15822          1         923    쿠폰적용사음품문의   NaN  Q     NaN   
15823          1         924    쿠폰적용사음품문의   NaN  Q     NaN   
15824          1         925    쿠폰적용사음품문의   NaN  Q     NaN   
15825          1         926    쿠폰적용사음품문의   NaN  Q     NaN   

                             MQ                   SQ     UA               SA  \
0              신발은 여기 있는 게 다예요?                  NaN    NaN              NaN   
1                           NaN                  NaN    NaN  네 성인이나 아동 다 있어요   
2                           NaN        발 사이즈 몇 신으세요?    NaN              NaN   
3                           NaN                  NaN  230이요              NaN   
4                           NaN  편하게 신을 수 있는 거 찾으세요?    NaN              NaN   
...                         ...                  ...    ...              ...   
15821              이 쿠폰에 적립해주세요                  NaN    NaN              NaN   
15822   사은품 받으려면 쿠폰 몇 개 모아야 해요?                  NaN    NaN              NaN   
15823  쿠폰 몇 개 모으면 사은품 받을 수 있어요?                  NaN    NaN              NaN   
15824  쿠폰 몇 장 모아야 사은품 받을 수 있나요?                  NaN    NaN              NaN   
15825   사은품 받으려면 쿠폰 몇 장 있어야 돼요?                  NaN    NaN              NaN   

          개체명 용어사전         지식베이스  Unnamed: 19  
0      신발, 여기  NaN         신발/제품          0.0  
1      성인, 아동  NaN  성인/대상, 아동/대상          NaN  
2      발, 사이즈  NaN           NaN          NaN  
3         230  NaN       230/사이즈          NaN  
4         NaN  NaN           NaN          NaN  
...       ...  ...           ...          ...  
15821     NaN  NaN           NaN          NaN  
15822     NaN  NaN           NaN          NaN  
15823     NaN  NaN           NaN          NaN  
15824     NaN  NaN           NaN          NaN  
15825     NaN  NaN           NaN          NaN  

[15826 rows x 20 columns]

In [6]:
df1.columns

Index(['SPEAKER', 'SENTENCE', 'DATAID', 'DOMAINID', 'DOMAIN', 'CATEGORY',
       'SPEAKERID', 'SENTENCEID', 'MAIN', 'SUB', 'QA', 'QACNCT', 'MQ', 'SQ',
       'UA', 'SA', '개체명', '용어사전', '지식베이스'],
      dtype='object')

In [7]:
df2.columns

Index(['SPEAKER', 'SENTENCE', 'DATAID', 'DOMAINID', 'DOMAIN', 'CATEGORY',
       'SPEAKERID', 'SENTENCEID', 'MAIN', 'SUB', 'QA', 'QACNCT', 'MQ', 'SQ',
       'UA', 'SA', '개체명', '용어사전', '지식베이스', 'Unnamed: 19'],
      dtype='object')

In [8]:
df2.drop('Unnamed: 19',axis=1, inplace = True)
df2.shape

(15826, 19)

In [9]:
len(df1)

15726

In [10]:
df1.tail()

SPEAKER                            SENTENCE DATAID DOMAINID DOMAIN  \
15721      고객              주문하려면 얼마 전에 연락드려야 하나요?     생성        A    음식점   
15722      고객  이번 목욜 애들 체육대회라 25개 미리 주문하려는데 가능하죠?     생성        A    음식점   
15723      고객               L 사이즈 면 어느 정도 되는 건가요?     생성        A    음식점   
15724      고객                 탕수육 제일 작은 사이즈는 뭐예요?     생성        A    음식점   
15725      고객                여기 피자사이즈가 피자헛하고 같아요?     생성        A    음식점   

      CATEGORY  SPEAKERID  SENTENCEID     MAIN  SUB QA QACNCT  \
15721    배달음식점          1         919   예약주문문의  NaN  Q    NaN   
15722    배달음식점          1         920   예약주문문의  NaN  Q    NaN   
15723    배달음식점          1         921  메뉴사이즈문의  NaN  Q    NaN   
15724    배달음식점          1         922  메뉴사이즈문의  NaN  Q    NaN   
15725    배달음식점          1         923  메뉴사이즈문의  NaN  Q    NaN   

                                       MQ   SQ   UA   SA  개체명 용어사전 지식베이스  
15721              주문하려면 얼마 전에 연락드려야 하나요?  NaN  NaN  NaN  NaN  NaN   NaN  
15722  이번 목욜 애들 체육대회라 25개 미리 주문하려는데 가능하죠?  NaN  NaN  NaN  NaN  NaN   NaN  
15723               L 사이즈 면 어느 정도 되는 건가요?  NaN  NaN  NaN  NaN  NaN   NaN  
15724                 탕수육 제일 작은 사이즈는 뭐예요?  NaN  NaN  NaN  NaN  NaN   NaN  
15725                여기 피자사이즈가 피자헛하고 같아요?  NaN  NaN  NaN  NaN  NaN   NaN

In [11]:
df = pd.concat([
    df1,df2,df3,df4,df5,df6,df7,df8
])

In [12]:
df[15725:15735]

SPEAKER                      SENTENCE      DATAID DOMAINID DOMAIN  \
15725      고객          여기 피자사이즈가 피자헛하고 같아요?          생성        A    음식점   
0          고객              신발은 여기 있는 게 다예요?  박은영_액세서리_5        B  의복의류점   
1          점원               네 성인이나 아동 다 있어요  박은영_액세서리_5        B  의복의류점   
2          점원                 발 사이즈 몇 신으세요?  박은영_액세서리_5        B  의복의류점   
3          고객                         230이요  박은영_액세서리_5        B  의복의류점   
4          점원           편하게 신을 수 있는 거 찾으세요?  박은영_액세서리_5        B  의복의류점   
5          고객          네 봄이니까 편하게 신을 수 있는 거  박은영_액세서리_5        B  의복의류점   
6          점원  이런 건 어떠세요? 이런 거도 신발 무척 편하거든요  박은영_액세서리_5        B  의복의류점   
7          고객                 굽 좀 높은 거 없나요?  박은영_액세서리_5        B  의복의류점   
8          점원     봄 상품은 아직 어른 제품이 많이 안나왔습니다  박은영_액세서리_5        B  의복의류점   

      CATEGORY  SPEAKERID  SENTENCEID         MAIN   SUB QA QACNCT  \
15725    배달음식점          1         923      메뉴사이즈문의   NaN  Q    NaN   
0           신발          1           1  종류별신발제품문의요청   NaN  Q    NaN   
1           신발          0           2  종류별신발제품문의요청   NaN  A    NaN   
2           신발          0           3  종류별신발제품문의요청   사이즈  Q    NaN   
3           신발          1           4  종류별신발제품문의요청   사이즈  A    NaN   
4           신발          0           5          착화감  제품문의  Q    NaN   
5           신발          1           6          착화감  제품문의  A    NaN   
6           신발          0           7          착화감  제품추천  Q    NaN   
7           신발          1           8        굽높이문의   NaN  Q    NaN   
8           신발          0           9        굽높이문의   NaN  A    NaN   

                         MQ                            SQ  \
15725  여기 피자사이즈가 피자헛하고 같아요?                           NaN   
0          신발은 여기 있는 게 다예요?                           NaN   
1                       NaN                           NaN   
2                       NaN                 발 사이즈 몇 신으세요?   
3                       NaN                           NaN   
4                       NaN           편하게 신을 수 있는 거 찾으세요?   
5                       NaN                           NaN   
6                       NaN  이런 건 어떠세요? 이런 거도 신발 무척 편하거든요   
7             굽 좀 높은 거 없나요?                           NaN   
8                       NaN                           NaN   

                         UA                         SA        개체명 용어사전  \
15725                   NaN                        NaN        NaN  NaN   
0                       NaN                        NaN     신발, 여기  NaN   
1                       NaN            네 성인이나 아동 다 있어요     성인, 아동  NaN   
2                       NaN                        NaN     발, 사이즈  NaN   
3                     230이요                        NaN        230  NaN   
4                       NaN                        NaN        NaN  NaN   
5      네 봄이니까 편하게 신을 수 있는 거                        NaN          봄  NaN   
6                       NaN                        NaN         신발  NaN   
7                       NaN                        NaN          굽  NaN   
8                       NaN  봄 상품은 아직 어른 제품이 많이 안나왔습니다  봄, 상품, 어른  NaN   

              지식베이스  
15725           NaN  
0             신발/제품  
1      성인/대상, 아동/대상  
2               NaN  
3           230/사이즈  
4               NaN  
5              봄/계절  
6             신발/제품  
7              굽/부분  
8       봄/계절, 어른/대상

In [13]:
df = pd.concat([
    df1,df2,df3,df4,df5,df6,df7,df8
], ignore_index=True)

In [14]:
df[15725:15735]

SPEAKER                      SENTENCE      DATAID DOMAINID DOMAIN  \
15725      고객          여기 피자사이즈가 피자헛하고 같아요?          생성        A    음식점   
15726      고객              신발은 여기 있는 게 다예요?  박은영_액세서리_5        B  의복의류점   
15727      점원               네 성인이나 아동 다 있어요  박은영_액세서리_5        B  의복의류점   
15728      점원                 발 사이즈 몇 신으세요?  박은영_액세서리_5        B  의복의류점   
15729      고객                         230이요  박은영_액세서리_5        B  의복의류점   
15730      점원           편하게 신을 수 있는 거 찾으세요?  박은영_액세서리_5        B  의복의류점   
15731      고객          네 봄이니까 편하게 신을 수 있는 거  박은영_액세서리_5        B  의복의류점   
15732      점원  이런 건 어떠세요? 이런 거도 신발 무척 편하거든요  박은영_액세서리_5        B  의복의류점   
15733      고객                 굽 좀 높은 거 없나요?  박은영_액세서리_5        B  의복의류점   
15734      점원     봄 상품은 아직 어른 제품이 많이 안나왔습니다  박은영_액세서리_5        B  의복의류점   

      CATEGORY  SPEAKERID  SENTENCEID         MAIN   SUB QA QACNCT  \
15725    배달음식점          1         923      메뉴사이즈문의   NaN  Q    NaN   
15726       신발          1           1  종류별신발제품문의요청   NaN  Q    NaN   
15727       신발          0           2  종류별신발제품문의요청   NaN  A    NaN   
15728       신발          0           3  종류별신발제품문의요청   사이즈  Q    NaN   
15729       신발          1           4  종류별신발제품문의요청   사이즈  A    NaN   
15730       신발          0           5          착화감  제품문의  Q    NaN   
15731       신발          1           6          착화감  제품문의  A    NaN   
15732       신발          0           7          착화감  제품추천  Q    NaN   
15733       신발          1           8        굽높이문의   NaN  Q    NaN   
15734       신발          0           9        굽높이문의   NaN  A    NaN   

                         MQ                            SQ  \
15725  여기 피자사이즈가 피자헛하고 같아요?                           NaN   
15726      신발은 여기 있는 게 다예요?                           NaN   
15727                   NaN                           NaN   
15728                   NaN                 발 사이즈 몇 신으세요?   
15729                   NaN                           NaN   
15730                   NaN           편하게 신을 수 있는 거 찾으세요?   
15731                   NaN                           NaN   
15732                   NaN  이런 건 어떠세요? 이런 거도 신발 무척 편하거든요   
15733         굽 좀 높은 거 없나요?                           NaN   
15734                   NaN                           NaN   

                         UA                         SA        개체명 용어사전  \
15725                   NaN                        NaN        NaN  NaN   
15726                   NaN                        NaN     신발, 여기  NaN   
15727                   NaN            네 성인이나 아동 다 있어요     성인, 아동  NaN   
15728                   NaN                        NaN     발, 사이즈  NaN   
15729                 230이요                        NaN        230  NaN   
15730                   NaN                        NaN        NaN  NaN   
15731  네 봄이니까 편하게 신을 수 있는 거                        NaN          봄  NaN   
15732                   NaN                        NaN         신발  NaN   
15733                   NaN                        NaN          굽  NaN   
15734                   NaN  봄 상품은 아직 어른 제품이 많이 안나왔습니다  봄, 상품, 어른  NaN   

              지식베이스  
15725           NaN  
15726         신발/제품  
15727  성인/대상, 아동/대상  
15728           NaN  
15729       230/사이즈  
15730           NaN  
15731          봄/계절  
15732         신발/제품  
15733          굽/부분  
15734   봄/계절, 어른/대상

In [15]:
df.shape

(82282, 19)

In [16]:
df.head(2)

SPEAKER   SENTENCE       DATAID DOMAINID DOMAIN CATEGORY  SPEAKERID  \
0      고객  지금 배달되나요?  박은영_전화_중식_1        A    음식점    배달음식점          1   
1      점원  아 네 배달됩니다  박은영_전화_중식_1        A    음식점    배달음식점          0   

   SENTENCEID    MAIN  SUB QA QACNCT         MQ   SQ   UA         SA  \
0           1  배달가능문의  NaN  Q    NaN  지금 배달되나요?  NaN  NaN        NaN   
1           2  배달가능문의  NaN  A    NaN        NaN  NaN  NaN  아 네 배달됩니다   

          개체명 용어사전 지식베이스  
0  수고, 지금, 배달  NaN   NaN  
1          배달  NaN   NaN

In [17]:
df.DOMAIN.unique()

array(['음식점', '의복의류점', '학원', '소매', '생활서비스', '카페', '숙박', '관광여가오락'],
      dtype=object)

In [18]:
df['SPEAKER'].unique()

array(['고객', '점원'], dtype=object)

In [19]:
df.groupby('DOMAIN').head()

SPEAKER                                           SENTENCE       DATAID  \
0          고객                                          지금 배달되나요?  박은영_전화_중식_1   
1          점원                                          아 네 배달됩니다  박은영_전화_중식_1   
2          고객                       짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?  박은영_전화_중식_1   
3          점원  특해물 짬뽕도 있고 전복 새우 짬뽕도 있고 해물 종류도 새우 홍합 전복 없는 게 없습니다  박은영_전화_중식_1   
4          고객                          전복 들어가는 거는 특해물 짬뽕 시켜야 돼요?  박은영_전화_중식_1   
15726      고객                                   신발은 여기 있는 게 다예요?   박은영_액세서리_5   
15727      점원                                    네 성인이나 아동 다 있어요   박은영_액세서리_5   
15728      점원                                      발 사이즈 몇 신으세요?   박은영_액세서리_5   
15729      고객                                              230이요   박은영_액세서리_5   
15730      점원                                편하게 신을 수 있는 거 찾으세요?   박은영_액세서리_5   
31552      고객                    애가 고등학교 1학년인데 태권도 하려면 수강료가 얼마쯤?    장미연_태권도_1   
31553      점원                                           12만 원입니다    장미연_태권도_1   
31554      점원                                      뭐 때문에 하시려는데요?    장미연_태권도_1   
31555      고객                        애가 겁도 많고 그래서 태권도 한 번 시켜보려고요    장미연_태권도_1   
31556      고객                                      고1이 몇 명쯤 있나요?    장미연_태권도_1   
36325      고객                                      삼겹살 1근에 얼마에요?     16335872   
36326      점원                                              만원입니다     16335872   
36327      고객                                  넷이 먹을건데 2근이면 되나요?     16335872   
36328      점원                                            네 충분하세요     16335872   
36329      고객                                           그럼 2근주세요     16335872   
51274      고객                                       첫 방문 할인 되나요?     16321031   
51275      점원                                       네, 30프로 됩니다.     16321031   
51276      고객                                 몇 시까지가야 펌 할 수 있나요?     16321031   
51277      점원                                  오후 6시까지는 오셔야 합니다.     16321031   
51278      고객                                 두피 케어 서비스는 얼마 인가요?     16321031   
62361      고객                                       아이스아메리카노 하나요     16334604   
62362      점원                                       테이크아웃하실 건가요?     16334604   
62363      고객                                       네 텀블러에 넣어주세요     16334604   
62364      점원                                  텀블러 할인 300원 해드릴게요     16334604   
62365      고객                                       그란데 사이즈로 주세요     16334604   
70220      고객                           사월 십삼일 일박 이일로 예약이 가능한가요?     박은영_호텔_1   
70221      점원                   네, 저희 홈페이지에 들어가면 지금 예약이 가능하실 거예요     박은영_호텔_1   
70222      고객                      숙박 비용이 대략 얼마인지 말씀을 해주실 수 있어요?     박은영_호텔_1   
70223      점원      사월 십삽일 같은 경우에는 극성수기 주말이라 스탠다드룸 두 분 기준에 구만원이에요     박은영_호텔_1   
70224      고객                                    사인 가족이 가면 얼마에요?     박은영_호텔_1   
77333      고객                                        1시간에 얼마인가요?     16531968   
77334      점원                처음 1시간은 1000원이고 이후 1시간은 500원씩 추가됩니다     16531968   
77335      고객                                         무인발급기 있나요?     16531968   
77336      점원                         무인발급기는 카운터 바로 옆쪽에 이용 가능합니다     16531968   
77337      고객                                   1달 정액권 끊을 수 있나요?     16531968   

      DOMAINID  DOMAIN CATEGORY  SPEAKERID  SENTENCEID               MAIN  \
0            A     음식점    배달음식점          1           1             배달가능문의   
1            A     음식점    배달음식점          0           2             배달가능문의   
2            A     음식점    배달음식점          1           3        베스트메뉴문의추천요청   
3            A     음식점    배달음식점          0           4        베스트메뉴문의추천요청   
4            A     음식점    배달음식점          1           5             메뉴정보문의   
15726        B   의복의류점       신발          1           1        종류별신발제품문의요청   
15727        B   의복의류점       신발          0           2        종류별신발제품문의요청   
15728        B   의복의류점       신발          0           3        종류별신발제

In [20]:
# 'groupby' 함수를 실행하고 SPEAKER (누가 말했는지 담겨있음.) row의 개수를 count
# 각 업종 별로 몇 개의 데이터가 있는지 확인할 수 있다.
df.groupby('DOMAIN').SPEAKER.count()

DOMAIN
관광여가오락     4949
생활서비스     11087
소매        14949
숙박         7113
음식점       15726
의복의류점     15826
카페         7859
학원         4773
Name: SPEAKER, dtype: int64

In [21]:
df[df.DOMAIN=='생활서비스'].shape

(11087, 19)

In [22]:
df[df.DOMAIN=='생활서비스'].groupby('CATEGORY').head()

SPEAKER                        SENTENCE     DATAID DOMAINID DOMAIN  \
51274      고객                    첫 방문 할인 되나요?   16321031        E  생활서비스   
51275      점원                    네, 30프로 됩니다.   16321031        E  생활서비스   
51276      고객              몇 시까지가야 펌 할 수 있나요?   16321031        E  생활서비스   
51277      점원               오후 6시까지는 오셔야 합니다.   16321031        E  생활서비스   
51278      고객              두피 케어 서비스는 얼마 인가요?   16321031        E  생활서비스   
51564      고객              소화 안 될 때 먹는 약 주세요.   16321032        E  생활서비스   
51565      점원                   네, 몇 개 필요하세요?   16321032        E  생활서비스   
51566      고객                         3개 주세요.   16321032        E  생활서비스   
51567      점원                  네, 잠시만 기다려주세요.   16321032        E  생활서비스   
51568      고객               혹시 미세먼지 마스크도 있나요?   16321032        E  생활서비스   
51834      고객           고등학생 남자인데 1인실은 얼마인가요?  장미연_독서실_1        E  생활서비스   
51835      점원                         16만원입니다  장미연_독서실_1        E  생활서비스   
51836      고객  16만원 1인실 어떻게 생겼는지 한 번 볼 수 있어요?  장미연_독서실_1        E  생활서비스   
51837      점원                        네, 따라오세요  장미연_독서실_1        E  생활서비스   
51838      고객                      다인실은 얼마에요?  장미연_독서실_1        E  생활서비스   
54934      고객           조끼 밍크는 세탁비가 얼마정도 하나요?  장미연_세탁소_1        E  생활서비스   
54935      점원                    밍크는 저희 안되는데요  장미연_세탁소_1        E  생활서비스   
54936      고객        사장님은 수거하고 갖다주고 그런건 안하세요?  장미연_세탁소_1        E  생활서비스   
54937      점원                       집이 어디신데요?  장미연_세탁소_1        E  생활서비스   
54938      고객                     상인 1동 이편한이요  장미연_세탁소_1        E  생활서비스   
58556      고객          가죽옷 자르는 것도 수선하실 수 있어요?  박청희_옷수선_1        E  생활서비스   
58557      점원                     네, 수선 가능합니다  박청희_옷수선_1        E  생활서비스   
58558      고객      길이가 어중간해서, 엉덩이까지 자를 수 있어요?  박청희_옷수선_1        E  생활서비스   
58559      점원                     네, 수선 가능합니다  박청희_옷수선_1        E  생활서비스   
58560      고객               사장님 가죽은 좀 비싸긴 하죠?  박청희_옷수선_1        E  생활서비스   
60038      고객                    여기 수선은 안 돼요?  김옥연_세탁소_1        E  생활서비스   
60039      점원                      수선은 안 합니다.  김옥연_세탁소_1        E  생활서비스   
60040      고객                여기 일반 세탁 다 받아줘요?  김옥연_세탁소_1        E  생활서비스   
60041      점원                               예  김옥연_세탁소_1        E  생활서비스   
60042      고객                         신발도 돼요?  김옥연_세탁소_1        E  생활서비스   

      CATEGORY  SPEAKERID  SENTENCEID               MAIN    SUB QA QACNCT  \
51274      미용실          1           1              할인 문의    NaN  Q    NaN   
51275      미용실          0           2              할인 문의    NaN  A    NaN   
51276      미용실          1           3  원하는 시간에 예약이 가능 문의    NaN  Q    NaN   
51277      미용실          0           4  원하는 시간에 예약이 가능 문의    NaN  A    NaN   
51278      미용실          1           5    모발 케어 서비스 가격 문의    NaN  Q    NaN   
51564       약국          1           1     처방전 외 약품 구입 요청    NaN  Q    NaN   
51565       약국          0           2     처방전 외 약품 구입 요청  구매 수량  Q    NaN   
51566       약국          1           3     처방전 외 약품 구입 요청  구매 수량  A    NaN   
51567       약국          0           4     처방전 외 약품 구입 요청    NaN  A      1   
51568       약국          1           5              제품 문의    NaN  Q    NaN   
51834      독서실          1           1              가격 문의    NaN  Q    NaN   
51835      독서실          0           2              가격 문의    NaN  A    NaN   
51836      독서실          1           3          열람실 보기 요청    NaN  Q    NaN   
51837      독서실          0           4          열람실 보기 요청    NaN  A    NaN   
51838      독서실          1           5              가격 문의    NaN  Q    NaN   
54934     세탁소           1           1              가격 문의    NaN  Q    NaN   
54935     세탁소           0           2              가격 문의    NaN  A    NaN   
54936     세탁소           1           3   세탁물 수거 및 배달가능 문의    NaN  Q    NaN   
54937     세탁소           0           4   세탁물 수거 및 배달가능 문의    NaN  Q    NaN   
54938     세탁소           1           5   세탁물 수거 및 배달가능 문의    NaN  A    NaN   
58556      옷수선          1       

In [23]:
df[df.DOMAIN=='생활서비스'].groupby('CATEGORY').SPEAKER.count()

CATEGORY
독서실      865
미용실     4318
세탁소      435
세탁소      980
약국      3365
옷수선     1124
Name: SPEAKER, dtype: int64

In [24]:
df.loc[df.DOMAIN=='생활서비스']

SPEAKER                   SENTENCE    DATAID DOMAINID DOMAIN CATEGORY  \
51274      고객               첫 방문 할인 되나요?  16321031        E  생활서비스      미용실   
51275      점원               네, 30프로 됩니다.  16321031        E  생활서비스      미용실   
51276      고객         몇 시까지가야 펌 할 수 있나요?  16321031        E  생활서비스      미용실   
51277      점원          오후 6시까지는 오셔야 합니다.  16321031        E  생활서비스      미용실   
51278      고객         두피 케어 서비스는 얼마 인가요?  16321031        E  생활서비스      미용실   
...       ...                        ...       ...      ...    ...      ...   
62356      고객           10만원 수표인데 결제되나요?        생성        E  생활서비스      옷수선   
62357      고객        2만5천원 현금로 결제하면 되나요?        생성        E  생활서비스      옷수선   
62358      고객            5만원권으로 결제해도 돼요?        생성        E  생활서비스      옷수선   
62359      고객  수선비 5천 원 현금으로만 결제할 수 있어요?        생성        E  생활서비스      옷수선   
62360      고객       현금으로 결제하면 좋은 점이 있나요?        생성        E  생활서비스      옷수선   

       SPEAKERID  SENTENCEID               MAIN  SUB QA QACNCT  \
51274          1           1              할인 문의  NaN  Q    NaN   
51275          0           2              할인 문의  NaN  A    NaN   
51276          1           3  원하는 시간에 예약이 가능 문의  NaN  Q    NaN   
51277          0           4  원하는 시간에 예약이 가능 문의  NaN  A    NaN   
51278          1           5    모발 케어 서비스 가격 문의  NaN  Q    NaN   
...          ...         ...                ...  ... ..    ...   
62356          1        2183           현금 결제 문의  NaN  Q    NaN   
62357          1        2184           현금 결제 문의  NaN  Q    NaN   
62358          1        2185           현금 결제 문의  NaN  Q    NaN   
62359          1        2186           현금 결제 문의  NaN  Q    NaN   
62360          1        2187           현금 결제 문의  NaN  Q    NaN   

                              MQ   SQ   UA                 SA        개체명 용어사전  \
51274                첫 방문 할인되나요?  NaN  NaN                NaN  첫, 방문, 할인  NaN   
51275                        NaN  NaN  NaN       네, 30프로 됩니다.       30프로  NaN   
51276        몇 시까지 가야 펌 할 수 있나요?  NaN  NaN                NaN     몇 시, 펌  NaN   
51277                        NaN  NaN  NaN  오후 6시까지는 오셔야 합니다.      오후 6시  NaN   
51278          두피 케어 서비스는 얼마인가요?  NaN  NaN                NaN  두피 케어 서비스  NaN   
...                          ...  ...  ...                ...        ...  ...   
62356           10만원 수표인데 결제되나요?  NaN  NaN                NaN        NaN  NaN   
62357        2만5천원 현금로 결제하면 되나요?  NaN  NaN                NaN        NaN  NaN   
62358            5만원권으로 결제해도 돼요?  NaN  NaN                NaN        NaN  NaN   
62359  수선비 5천 원 현금으로만 결제할 수 있어요?  NaN  NaN                NaN        NaN  NaN   
62360       현금으로 결제하면 좋은 점이 있나요?  NaN  NaN                NaN        NaN  NaN   

              지식베이스  
51274           NaN  
51275      30프로/할인율  
51276          펌/시술  
51277      오후 6시/시간  
51278  두피 케어 서비스/시술  
...             ...  
62356           NaN  
62357           NaN  
62358           NaN  
62359           NaN  
62360           NaN  

[11087 rows x 19 columns]

In [25]:
df.loc[(df.DOMAIN=='생활서비스') & (df.CATEGORY=='미용실')] #총 4138개

SPEAKER                     SENTENCE    DATAID DOMAINID DOMAIN CATEGORY  \
51274      고객                 첫 방문 할인 되나요?  16321031        E  생활서비스      미용실   
51275      점원                 네, 30프로 됩니다.  16321031        E  생활서비스      미용실   
51276      고객           몇 시까지가야 펌 할 수 있나요?  16321031        E  생활서비스      미용실   
51277      점원            오후 6시까지는 오셔야 합니다.  16321031        E  생활서비스      미용실   
51278      고객           두피 케어 서비스는 얼마 인가요?  16321031        E  생활서비스      미용실   
...       ...                          ...       ...      ...    ...      ...   
61396      고객  혼자해도 효과 좋은 케어 방법 좀 알려주시겠어요?        생성        E  생활서비스      미용실   
61397      고객            집에서는 어떻게 관리하면 돼요?        생성        E  생활서비스      미용실   
61398      고객  집에서 쉽게 할 수 있는 케어 방법 좀 알려주세요        생성        E  생활서비스      미용실   
61399      고객             홈 케어는 어떻게 하면 돼요?        생성        E  생활서비스      미용실   
61400      고객        집에서는 에센스만 잘 발라주면 될까요?        생성        E  생활서비스      미용실   

       SPEAKERID  SENTENCEID               MAIN  SUB QA QACNCT  \
51274          1           1              할인 문의  NaN  Q    NaN   
51275          0           2              할인 문의  NaN  A    NaN   
51276          1           3  원하는 시간에 예약이 가능 문의  NaN  Q    NaN   
51277          0           4  원하는 시간에 예약이 가능 문의  NaN  A    NaN   
51278          1           5    모발 케어 서비스 가격 문의  NaN  Q    NaN   
...          ...         ...                ...  ... ..    ...   
61396          1        1223         홈 케어 방법 문의  NaN  Q    NaN   
61397          1        1224         홈 케어 방법 문의  NaN  Q    NaN   
61398          1        1225         홈 케어 방법 문의  NaN  Q    NaN   
61399          1        1226         홈 케어 방법 문의  NaN  Q    NaN   
61400          1        1227         홈 케어 방법 문의  NaN  Q    NaN   

                                MQ   SQ   UA                 SA        개체명  \
51274                  첫 방문 할인되나요?  NaN  NaN                NaN  첫, 방문, 할인   
51275                          NaN  NaN  NaN       네, 30프로 됩니다.       30프로   
51276          몇 시까지 가야 펌 할 수 있나요?  NaN  NaN                NaN     몇 시, 펌   
51277                          NaN  NaN  NaN  오후 6시까지는 오셔야 합니다.      오후 6시   
51278            두피 케어 서비스는 얼마인가요?  NaN  NaN                NaN  두피 케어 서비스   
...                            ...  ...  ...                ...        ...   
61396  혼자해도 효과 좋은 케어 방법 좀 알려주시겠어요?  NaN  NaN                NaN        NaN   
61397            집에서는 어떻게 관리하면 돼요?  NaN  NaN                NaN        NaN   
61398  집에서 쉽게 할 수 있는 케어 방법 좀 알려주세요  NaN  NaN                NaN        NaN   
61399             홈 케어는 어떻게 하면 돼요?  NaN  NaN                NaN        NaN   
61400        집에서는 에센스만 잘 발라주면 될까요?  NaN  NaN                NaN        NaN   

      용어사전         지식베이스  
51274  NaN           NaN  
51275  NaN      30프로/할인율  
51276  NaN          펌/시술  
51277  NaN      오후 6시/시간  
51278  NaN  두피 케어 서비스/시술  
...    ...           ...  
61396  NaN           NaN  
61397  NaN           NaN  
61398  NaN           NaN  
61399  NaN           NaN  
61400  NaN           NaN  

[4318 rows x 19 columns]

#### 미용실 관련 대화 data는 총 4318개
- 이 중 질문으로 이루어진 데이터는 2672, 답변으로 이루어진 데이터는 1646 개임
- 데이터의 질문도 답도 결국 반복학습을 통해 정확도를 높이기 때문에 미용실과 관련된 질문 뿐만 아니라 타 카테고리의 질문/답변 set 도 가져옴
- 따라서, main 컬럼의 value 중 '문의', '추천', '요청' 등 테마별로 나눠서 추가적으로 데이터 가져옴


In [26]:
df.loc[(df.DOMAIN=='생활서비스') & (df.CATEGORY=='미용실') & (df.QA=='Q')]

SPEAKER                     SENTENCE    DATAID DOMAINID DOMAIN CATEGORY  \
51274      고객                 첫 방문 할인 되나요?  16321031        E  생활서비스      미용실   
51276      고객           몇 시까지가야 펌 할 수 있나요?  16321031        E  생활서비스      미용실   
51278      고객           두피 케어 서비스는 얼마 인가요?  16321031        E  생활서비스      미용실   
51280      고객                 염색약은 뭐 쓰시나요?  16321031        E  생활서비스      미용실   
51282      고객              커트하고 머리 감겨 주나요?  16321031        E  생활서비스      미용실   
...       ...                          ...       ...      ...    ...      ...   
61396      고객  혼자해도 효과 좋은 케어 방법 좀 알려주시겠어요?        생성        E  생활서비스      미용실   
61397      고객            집에서는 어떻게 관리하면 돼요?        생성        E  생활서비스      미용실   
61398      고객  집에서 쉽게 할 수 있는 케어 방법 좀 알려주세요        생성        E  생활서비스      미용실   
61399      고객             홈 케어는 어떻게 하면 돼요?        생성        E  생활서비스      미용실   
61400      고객        집에서는 에센스만 잘 발라주면 될까요?        생성        E  생활서비스      미용실   

       SPEAKERID  SENTENCEID               MAIN  SUB QA QACNCT  \
51274          1           1              할인 문의  NaN  Q    NaN   
51276          1           3  원하는 시간에 예약이 가능 문의  NaN  Q    NaN   
51278          1           5    모발 케어 서비스 가격 문의  NaN  Q    NaN   
51280          1           7           염색 제품 문의  NaN  Q    NaN   
51282          1           9          샴푸 서비스 문의  NaN  Q    NaN   
...          ...         ...                ...  ... ..    ...   
61396          1        1223         홈 케어 방법 문의  NaN  Q    NaN   
61397          1        1224         홈 케어 방법 문의  NaN  Q    NaN   
61398          1        1225         홈 케어 방법 문의  NaN  Q    NaN   
61399          1        1226         홈 케어 방법 문의  NaN  Q    NaN   
61400          1        1227         홈 케어 방법 문의  NaN  Q    NaN   

                                MQ   SQ   UA   SA        개체명 용어사전  \
51274                  첫 방문 할인되나요?  NaN  NaN  NaN  첫, 방문, 할인  NaN   
51276          몇 시까지 가야 펌 할 수 있나요?  NaN  NaN  NaN     몇 시, 펌  NaN   
51278            두피 케어 서비스는 얼마인가요?  NaN  NaN  NaN  두피 케어 서비스  NaN   
51280                 염색약은 뭐 쓰시나요?  NaN  NaN  NaN        염색약  NaN   
51282              커트하고 머리 감겨 주나요?  NaN  NaN  NaN     커트, 머리  NaN   
...                            ...  ...  ...  ...        ...  ...   
61396  혼자해도 효과 좋은 케어 방법 좀 알려주시겠어요?  NaN  NaN  NaN        NaN  NaN   
61397            집에서는 어떻게 관리하면 돼요?  NaN  NaN  NaN        NaN  NaN   
61398  집에서 쉽게 할 수 있는 케어 방법 좀 알려주세요  NaN  NaN  NaN        NaN  NaN   
61399             홈 케어는 어떻게 하면 돼요?  NaN  NaN  NaN        NaN  NaN   
61400        집에서는 에센스만 잘 발라주면 될까요?  NaN  NaN  NaN        NaN  NaN   

              지식베이스  
51274           NaN  
51276          펌/시술  
51278  두피 케어 서비스/시술  
51280        염색약/제품  
51282  커트/시술, 머리/부위  
...             ...  
61396           NaN  
61397           NaN  
61398           NaN  
61399           NaN  
61400           NaN  

[2672 rows x 19 columns]

In [27]:
df[df.DOMAIN == '생활서비스'].MAIN.unique()

array(['할인 문의', '원하는 시간에 예약이 가능 문의', '모발 케어 서비스 가격 문의', '염색 제품 문의',
       '샴푸 서비스 문의', '커트 가격 문의', '선생님별 스케줄 문의', '커트 문의', '두피 케어 서비스 가격 문의',
       '결제 요청', '염색 문의', '염색 가격 문의', '동시 시술 문의', '동시 시술 요청', '펌 문의',
       '시술 시 손상 정도 문의', '염색 요청', '대기시간 문의', '선생님 요청',
       '사진 제시하면서 동일한 펌 가능한 지 문의', '책 잡지 신문 서비스 가능 문의', '헤어스타일 문의',
       '커트 요청', '참고 사진 제시하며 스타일 요청', '예약 문의', '펌 가격 문의', '소요시간 문의',
       '예약 요청', '펌 요청', '가격 문의', '오픈시간 문의', '추천 펌 문의', '기장 추가 가격 문의',
       '예약시간 문의', nan, '탈색 문의', '염색 전 탈색 문의', '탈색 횟수 문의', '커트 문의 대기 요구',
       '시술 중 서비스 가능 문의', '카카오페이 가능 여부 문의', '삼성페이 가능 여부 문의', '매직 요청',
       '현재 본인 스타일의 염색 색상 추천 요청', '추천 커트 문의', '음료 요청', '드라이 및 스타일링',
       '처방전 외 약품 구입 요청', '제품 문의', '의약품 구매 요청', '의약품 요청', '특정 제품 요청',
       '정수기 위치 문의', '휴일 문의', '약품 처리 요청', '단순 구매 요청', '페이 결제 요청',
       '영수증 비요청', '의약품 추천 요청', '제품 형태에 따른 의약품 요청', '유의사항 문의',
       '특정 증상 의약품 요청', '현금영수증 요청', '복용방법 문의', '전문가 의견 문의', '카드 결제 요청',
       '증상 문의', '제품 성분 문의', '오픈 시간 문의', '처방전 외 약품 구입

In [28]:
df1=df[df.MAIN.str.contains('문의')==True]
df1.shape

(60695, 19)

In [29]:
df1.head() #문의와 관련된 df

SPEAKER                                           SENTENCE       DATAID  \
0      고객                                          지금 배달되나요?  박은영_전화_중식_1   
1      점원                                          아 네 배달됩니다  박은영_전화_중식_1   
2      고객                       짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?  박은영_전화_중식_1   
3      점원  특해물 짬뽕도 있고 전복 새우 짬뽕도 있고 해물 종류도 새우 홍합 전복 없는 게 없습니다  박은영_전화_중식_1   
4      고객                          전복 들어가는 거는 특해물 짬뽕 시켜야 돼요?  박은영_전화_중식_1   

  DOMAINID DOMAIN CATEGORY  SPEAKERID  SENTENCEID         MAIN  SUB QA QACNCT  \
0        A    음식점    배달음식점          1           1       배달가능문의  NaN  Q    NaN   
1        A    음식점    배달음식점          0           2       배달가능문의  NaN  A    NaN   
2        A    음식점    배달음식점          1           3  베스트메뉴문의추천요청  NaN  Q    NaN   
3        A    음식점    배달음식점          0           4  베스트메뉴문의추천요청  NaN  A    NaN   
4        A    음식점    배달음식점          1           5       메뉴정보문의  NaN  Q    NaN   

                             MQ   SQ   UA  \
0                     지금 배달되나요?  NaN  NaN   
1                           NaN  NaN  NaN   
2  짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?  NaN  NaN   
3                           NaN  NaN  NaN   
4     전복 들어가는 거는 특해물 짬뽕 시켜야 돼요?  NaN  NaN   

                                                  SA  \
0                                                NaN   
1                                          아 네 배달됩니다   
2                                                NaN   
3  특해물 짬뽕도 있고 전복 새우 짬뽕도 있고 해물 종류도 새우 홍합 전복 없는 게 없습니다   
4                                                NaN   

                                   개체명 용어사전  \
0                           수고, 지금, 배달  NaN   
1                                   배달  NaN   
2                                   짬뽕  NaN   
3  특해물 짬뽕, 전복 새우 짬뽕, 해물 종류, 새우, 홍합, 전복  NaN   
4                           전복, 특해물 짬뽕  NaN   

                                               지식베이스  
0                                                NaN  
1                                                NaN  
2                                              짬뽕/메뉴  
3  특해물 짬뽕/메뉴, 전복 새우 짬뽕/메뉴, 새우/해물/재료, 홍합/해물/재료, 전복...  
4                                전복/해물/재료, 특해물 짬뽕/메뉴

In [30]:
df_que = df1[df1['QA']=='Q']
df_que

SPEAKER                      SENTENCE       DATAID DOMAINID  DOMAIN  \
0          고객                     지금 배달되나요?  박은영_전화_중식_1        A     음식점   
2          고객  짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?  박은영_전화_중식_1        A     음식점   
4          고객     전복 들어가는 거는 특해물 짬뽕 시켜야 돼요?  박은영_전화_중식_1        A     음식점   
6          고객    전복 들어가고 여러 가지 또 딴 것도 들어가죠?  박은영_전화_중식_1        A     음식점   
8          고객          마차이 짬뽕밥은 돼지고기 들어가나요?  박은영_전화_중식_1        A     음식점   
...       ...                           ...          ...      ...     ...   
82270      고객      포켓볼 당구대는 그냥 바로 가서 치면 돼요?           생성        H  관광여가오락   
82275      고객                  당구장 쿠폰이 있나요?           생성        H  관광여가오락   
82276      고객         여기서 하는 적립 쿠폰 같은거 있어요?           생성        H  관광여가오락   
82277      고객         당구장에서 자체적으로 쿠폰 만든거예요?           생성        H  관광여가오락   
82278      고객               여기서도 이 쿠폰이 되나요?           생성        H  관광여가오락   

      CATEGORY  SPEAKERID  SENTENCEID         MAIN  SUB QA QACNCT  \
0        배달음식점          1           1       배달가능문의  NaN  Q    NaN   
2        배달음식점          1           3  베스트메뉴문의추천요청  NaN  Q    NaN   
4        배달음식점          1           5       메뉴정보문의  NaN  Q    NaN   
6        배달음식점          1           7       메뉴정보문의  NaN  Q    NaN   
8        배달음식점          1           9       메뉴정보문의  NaN  Q    NaN   
...        ...        ...         ...          ...  ... ..    ...   
82270      당구장          1         234     이용 방법 문의  NaN  Q    NaN   
82275      당구장          1         239     쿠폰 유무 문의  NaN  Q    NaN   
82276      당구장          1         240     쿠폰 유무 문의  NaN  Q    NaN   
82277      당구장          1         241     쿠폰 유무 문의  NaN  Q    NaN   
82278      당구장          1         242     쿠폰 유무 문의  NaN  Q    NaN   

                                 MQ   SQ   UA   SA            개체명 용어사전  \
0                         지금 배달되나요?  NaN  NaN  NaN     수고, 지금, 배달  NaN   
2      짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?  NaN  NaN  NaN             짬뽕  NaN   
4         전복 들어가는 거는 특해물 짬뽕 시켜야 돼요?  NaN  NaN  NaN     전복, 특해물 짬뽕  NaN   
6        전복 들어가고 여러 가지 또 딴 것도 들어가죠?  NaN  NaN  NaN             전복  NaN   
8              마차이 짬뽕밥은 돼지고기 들어가나요?  NaN  NaN  NaN  마차이 짬뽕밥, 돼지고기  NaN   
...                             ...  ...  ...  ...            ...  ...   
82270      포켓볼 당구대는 그냥 바로 가서 치면 돼요?  NaN  NaN  NaN            NaN  NaN   
82275                  당구장 쿠폰이 있나요?  NaN  NaN  NaN            NaN  NaN   
82276         여기서 하는 적립 쿠폰 같은거 있어요?  NaN  NaN  NaN            NaN  NaN   
82277         당구장에서 자체적으로 쿠폰 만든거예요?  NaN  NaN  NaN            NaN  NaN   
82278               여기서도 이 쿠폰이 되나요?  NaN  NaN  NaN            NaN  NaN   

                     지식베이스  
0                      NaN  
2                    짬뽕/메뉴  
4      전복/해물/재료, 특해물 짬뽕/메뉴  
6                 전복/해물/재료  
8      마차이 짬뽕밥/메뉴, 돼지고기/재료  
...                    ...  
82270                  NaN  
82275                  NaN  
82276                  NaN  
82277                  NaN  
82278                  NaN  

[33533 rows x 19 columns]

In [31]:
df_que.shape
#문의과 관련된 df 중 질문 내용만

(33533, 19)

In [32]:
df_que.SENTENCE

0                           지금 배달되나요?
2        짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?
4           전복 들어가는 거는 특해물 짬뽕 시켜야 돼요?
6          전복 들어가고 여러 가지 또 딴 것도 들어가죠?
8                마차이 짬뽕밥은 돼지고기 들어가나요?
                     ...             
82270        포켓볼 당구대는 그냥 바로 가서 치면 돼요?
82275                    당구장 쿠폰이 있나요?
82276           여기서 하는 적립 쿠폰 같은거 있어요?
82277           당구장에서 자체적으로 쿠폰 만든거예요?
82278                 여기서도 이 쿠폰이 되나요?
Name: SENTENCE, Length: 33533, dtype: object

In [33]:
df_que.to_csv('./df_que.csv', encoding='utf-8-sig')

In [34]:
#위와 같은 방식으로 '추천' 과 '요청'의 질문 내용이 담긴 df따로 만들기 

In [35]:
df2=df[df.MAIN.str.contains('추천')==True]
df2.shape

(2495, 19)

In [36]:
df_rec = df2[df2['QA']=='Q']
df_rec
#추천과 관련된 df 중 질문 내용만

SPEAKER                           SENTENCE       DATAID DOMAINID DOMAIN  \
2          고객       짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?  박은영_전화_중식_1        A    음식점   
14         고객                     주로 어떤 게 잘 나가요?  박은영_전화_중식_2        A    음식점   
26         고객  그러면 4인 가족이면 탕수육 하나에 딴 거 어떤 거 시켜요?  박은영_전화_중식_2        A    음식점   
59         고객           짬뽕이나 간짜장 중에 어떤 게 잘 나가나요?  박은영_전화_중식_4        A    음식점   
71         고객             그러면 주로 어떤 거 시키나요 사람들이?  박은영_전화_중식_5        A    음식점   
...       ...                                ...          ...      ...    ...   
70054      고객               과일 들어간 음료 하나 추천해 주세요           생성        F     카페   
70055      고객                 가장 잘 나가는 음료 추천해주세요           생성        F     카페   
70056      고객             시원한 음료 중 뭐가 제일 잘 나가나요?           생성        F     카페   
70057      고객               많이 달지 않은 케익 좀 추천해주세요           생성        F     카페   
70058      고객      풍미가 좋은 커피 원두로 만든 커피 추천 가능할까요?           생성        F     카페   

      CATEGORY  SPEAKERID  SENTENCEID         MAIN  SUB QA QACNCT  \
2        배달음식점          1           3  베스트메뉴문의추천요청  NaN  Q    NaN   
14       배달음식점          1           3  베스트메뉴문의추천요청  NaN  Q    NaN   
26       배달음식점          1          15  취향에따른메뉴추천요구  NaN  Q    NaN   
59       배달음식점          1           3  베스트메뉴문의추천요청  NaN  Q    NaN   
71       배달음식점          1           7  베스트메뉴문의추천요청  NaN  Q    NaN   
...        ...        ...         ...          ...  ... ..    ...   
70054       카페          1         514       메뉴추천요구  NaN  Q    NaN   
70055       카페          1         515       메뉴추천요구  NaN  Q    NaN   
70056       카페          1         516       메뉴추천요구  NaN  Q    NaN   
70057       카페          1         517       메뉴추천요구  NaN  Q    NaN   
70058       카페          1         518       메뉴추천요구  NaN  Q    NaN   

                                      MQ   SQ   UA   SA             개체명 용어사전  \
2           짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?  NaN  NaN  NaN              짬뽕  NaN   
14                        주로 어떤 게 잘 나가요?  NaN  NaN  NaN             NaN  NaN   
26     그러면 4인 가족이면 탕수육 하나에 딴 거 어떤 거 시켜요?  NaN  NaN  NaN  4인 가족, 탕수육, 하나  NaN   
59              짬뽕이나 간짜장 중에 어떤 게 잘 나가나요?  NaN  NaN  NaN         짬뽕, 간짜장  NaN   
71                그러면 주로 어떤 거 시키나요 사람들이?  NaN  NaN  NaN             사람들  NaN   
...                                  ...  ...  ...  ...             ...  ...   
70054               과일 들어간 음료 하나 추천해 주세요  NaN  NaN  NaN             NaN  NaN   
70055                 가장 잘 나가는 음료 추천해주세요  NaN  NaN  NaN             NaN  NaN   
70056             시원한 음료 중 뭐가 제일 잘 나가나요?  NaN  NaN  NaN             NaN  NaN   
70057               많이 달지 않은 케익 좀 추천해주세요  NaN  NaN  NaN             NaN  NaN   
70058      풍미가 좋은 커피 원두로 만든 커피 추천 가능할까요?  NaN  NaN  NaN             NaN  NaN   

                         지식베이스  
2                        짬뽕/메뉴  
14                         NaN  
26     4인 가족/인원, 탕수육/메뉴, 하나/수량  
59               짬뽕/메뉴, 간짜장/메뉴  
71                         NaN  
...                        ...  
70054                      NaN  
70055                      NaN  
70056                      NaN  
70057                      NaN  
70058                      NaN  

[1378 rows x 19 columns]

In [37]:
df_rec.reset_index(drop=True, inplace=True) #reset_index(drop=False, inplace=False) 가 디폴트

In [38]:
df_rec.shape

(1378, 19)

In [39]:
len(df_rec['SENTENCE'])

1378

In [40]:
df_rec.to_csv('./df_rec.csv', encoding='utf-8-sig')

In [41]:
# df3=df[df.MAIN.str.contains('요청')==True]
# df3.shape

In [42]:
# df_req = df3[df3['QA']=='Q']
# df_req

In [43]:
# df_req.shape

In [44]:
# df_req['MAIN'].unique()

In [45]:
# df_req['SENTENCE'][:5000]

In [46]:
# df_req.to_csv('./df_req.csv', encoding='utf-8-sig') 

- 추천/ 요청 / 문의
- df_rec, df_req, df_que 로 각 질문 카테고리별 df 파일 생성.
- 하지만 "짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?" 처럼 
- main 컬럼에 베스트 메뉴 문의 추천 요청 과 같이 모든 카테고리에 해당된 문장 존재
- 중복 삭제 처리를 하게 될 경우 , "짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?" 에선 문의 혹은 추천 혹은 요청 하나만 남게됨
- 따라서 중복 삭제 처리는 하지 않고 목적에 따른 카테고리 분류만 진행하고자 함

In [47]:
# df_req['SENTENCE']

In [48]:
df_que['SENTENCE']

0                           지금 배달되나요?
2        짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?
4           전복 들어가는 거는 특해물 짬뽕 시켜야 돼요?
6          전복 들어가고 여러 가지 또 딴 것도 들어가죠?
8                마차이 짬뽕밥은 돼지고기 들어가나요?
                     ...             
82270        포켓볼 당구대는 그냥 바로 가서 치면 돼요?
82275                    당구장 쿠폰이 있나요?
82276           여기서 하는 적립 쿠폰 같은거 있어요?
82277           당구장에서 자체적으로 쿠폰 만든거예요?
82278                 여기서도 이 쿠폰이 되나요?
Name: SENTENCE, Length: 33533, dtype: object

 - 문의, 추천 질문 대화 셋만 쓴다고 했을 때 요청 대화셋에서도 쓸만한 내용이 있기 때문에 문의와 요청 대화 셋을 합쳐서 문의로 만들어 쓰려 했으나, 실제 오픈을 해보니 미용실 입장서 점원과 손님의 대화에 어울리는 상황이 많지 정보 제공을 하는 쳇봇(우리)과 일반 사용자 간의 대화내용에는 적합하지 않음

In [49]:
df.loc[(df.DOMAIN=='생활서비스') & (df.CATEGORY=='미용실') & (df.QA=='Q') & (df.MAIN.str.contains('요청')) 
       & (df.SPEAKER=='고객')].SENTENCE[50:100]

52612          저번에 머리해주셨던 선생님은 오늘 안 계시네요?
52614          그럼 혹시 다른 파마 잘 하는 선생님 계시나요?
52636                         적당하게 다듬어주세요
52638                           가위로만 해주세요
52640                        옆머리는 짧게 해주세요
52670                          그럼 한번 해볼게요
52690                  시원한 커피도 한 잔 부탁드릴게요
52692                    이 사진처럼 같은 머리해주세요
52700                         네 조금만 잘라주세요
52710        그럼 오늘은 일단 앞머리 커트하고 다음 주에 올게요
52714               그럼 머리 좀 자르고 세팅 파마 할게요
52752                         단발로 자르고 싶어요
52758                    네 짙은 갈색으로 염색해주세요
52760                            예쁘게 해주세요
52770                    그럼 영양이랑 볼륨매직해주세요
52778                      그럼 갈색으로 염색해주세요
52802                              커트해주세요
52804                          섀기커트로 해주세요
52834                   일단 앞머리는 눈썹 위로해주세요
52836                    전체적으로 조금씩만 잘라주세요
52846                         마실 것 좀 있나요?
52848                             녹차로 주세요
52860              그럼 파마하고 나서 염색도 같이 해주세요
52870                      그러면 좀 작

In [52]:
df_que_test = pd.concat([df_que], ignore_index=True)
df_que_test[300:350]

SPEAKER                                 SENTENCE       DATAID DOMAINID  \
300      고객                       중간 사이즈 두 개짜리도 있나요?  김옥연_전화_피자_1        A   
301      고객                    피자는 종류가 고기 종류로 나눠지나요?  김옥연_전화_피자_1        A   
302      고객                             이게 콤비네이션인가요?  김옥연_전화_피자_1        A   
303      고객                             피자 치즈는 뭐 써요?  김옥연_전화_피자_1        A   
304      고객                                 치킨도 있어요?  김옥연_전화_피자_1        A   
305      고객                               어떤 게 맛있어요?  김옥연_전화_피자_1        A   
306      점원                            몇 분이서 드실 겁니까?  김옥연_전화_피자_1        A   
307      고객                             이것도 단품도 있나요?  김옥연_전화_피자_1        A   
308      점원                                배달료가 있습니다  김옥연_전화_피자_1        A   
309      고객                                    얼마에요?  김옥연_전화_피자_1        A   
310      고객                         결제는 미리 하고 갈 건데요?  김옥연_전화_피자_1        A   
311      고객                            그럼 여기서는 얼만데요?  김옥연_전화_피자_1        A   
312      고객                                 그럼 레귤러는?  김옥연_전화_피자_1        A   
313      점원                           몇 시까지 갖다 드릴까요?  김옥연_전화_피자_1        A   
314      고객                      큐브 피자인가? 새로 나온 거예요?  이수연_전화_피자_1        A   
315      고객                              배달료 따로 받아요?  이수연_전화_피자_1        A   
316      고객                                    얼마에요?  이수연_전화_피자_1        A   
317      고객               낮에 시켜야 되거든요, 배달 시간은 언제 돼요?  이수연_전화_피자_1        A   
318      고객                               가격은 얼마인가요?  이수연_전화_피자_2        A   
319      고객                    아 그러면 위에 토핑이 뭐가 올라가요?  이수연_전화_피자_2        A   
320      고객                               애들이 잘 먹어요?  이수연_전화_피자_2        A   
321      점원                        애들이 연령대가 어떻게 됩니까?  이수연_전화_피자_2        A   
322      고객                          아 그러면 배달료는 받나요?  이수연_전화_피자_2        A   
323      고객                   치즈 크러스트 위에 토핑이 뭐가 있어요?  이수연_전화_피자_2        A   
324      고객                                그거는 가격대가?  이수연_전화_피자_2        A   
325      고객                                몇 시부터 해요?  이수연_전화_피자_2        A   
326      고객                  그거는 위에 토핑이 뭐가 들어가는 거예요?  이수연_전화_피자_3        A   
327      고객                        근데 초등학생 애들 잘 먹나요?  이수연_전화_피자_3        A   
328      고객  컵 같은 데다가 메인 메뉴 말고 사이드 메뉴같이 하나 있는 거 같던데?  이수연_전화_피자_3        A   
329      고객                           배달은 몇 시부터 합니까?  이수연_전화_피자_3        A   
330      고객               배달료를 따로 받던데 매장에서 받으면 다른가요?  이수연_전화_피자_3        A   
331      고객                       근데 사십프로 할인하는 거 맞죠?  장미연_전화_피자_1        A   
332      고객              그러면 고구마 피자 말고 다른 거 뭐 잘 나가요?  장미연_전화_피자_1        A   
333      고객                   아 그럼 거기 월성 보성 이 단지 맞죠?  장미연_전화_피자_1        A   
334      고객                        나가면 얼마나 기다려야 되나요?  장미연_전화_피자_1        A   
335      고객                      네 그 반도 유보란데 배달 되나요?  장미연_전화_피자_2        A   
336      점원                                무슨 동이십니까?  장미연_전화_피자_2        A   
337      고객                       그리고 40프로 하는게 끝났나요?  장미연_전화_피자_2        A   
338      고객                      동구는 그러면 어디로 걸어야 돼요?  장미연_전화_피자_2        A   
339      고객             그러면 위치 좀 가르쳐 주시면 내일이나 사러 갈게요  장미연_전화_피자_2        A   
340      고객                           국물 없는 짬뽕은 뭐에요?  최연정_전화_중식_1        A   
341      고객                               지금 바로 오시죠?  최연정_전화_중식_1        A   
342      고객                                 지금 배달돼요?  최연정_전화_중식_2        A   
343      고객                    간짜장하고 짜장면하고 차이점이 뭐에요?  최연정_전화_중식_2        A   
344      고객                   그러면 간짜장이 더 비싸겠는데 얼마에요?  최연정_전화_중식_2        A   
345      고객                                 그냥 짜장은요?  최연정_전화_중식_2        A   
346      고객                     지금 시키면 언제까지 올 수 있어요?  최연정_전화_중식_2        A   
347      고객                   그러면 한 10분 안에 올 수 있는거죠?  최연정_전화_중식_2        A   
348      고객                                외동 배달되나요?  김옥연_전화_중식_1        A   
349      고객                               요리종류도 있어요?  김옥연_전화_중식_1        A   

    DOMAIN CATEGORY  SPEA

In [53]:
# df_que_test.info()

In [54]:
# df_que_test.isnull().sum()

In [55]:
# df_que_test['SENTENCE'].drop_duplicates()

In [56]:
# len(df_que_test['SENTENCE'].drop_duplicates())

In [57]:
# df_que_test.index=range(len(df_que_test))

In [58]:
# df_que_test.info()

In [59]:
# df_que = df_que_test

In [60]:
# df_que['SENTENCE']

In [61]:
# df_que[:30]

In [62]:
# df_rec.head()

In [63]:
# len(df_rec)

## => df_que 는 문의 대화 dataset 의도만 담긴 df , df_rec는 추천 대화 dataset 중 질문과 그 의도만 담긴 df로 따로 만들자

In [64]:
df_combine = pd.concat([df_que, df_rec], ignore_index=True)
df_combine.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34911 entries, 0 to 34910
Data columns (total 19 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   SPEAKER     34911 non-null  object
 1   SENTENCE    34911 non-null  object
 2   DATAID      34911 non-null  object
 3   DOMAINID    34911 non-null  object
 4   DOMAIN      34911 non-null  object
 5   CATEGORY    34911 non-null  object
 6   SPEAKERID   34911 non-null  int64 
 7   SENTENCEID  34911 non-null  int64 
 8   MAIN        34911 non-null  object
 9   SUB         3470 non-null   object
 10  QA          34911 non-null  object
 11  QACNCT      1 non-null      object
 12  MQ          31373 non-null  object
 13  SQ          3537 non-null   object
 14  UA          1 non-null      object
 15  SA          0 non-null      object
 16  개체명         27463 non-null  object
 17  용어사전        1229 non-null   object
 18  지식베이스       18382 non-null  object
dtypes: int64(2), object(17)
memory usage: 5.1+ MB


In [65]:
df_combine[50:100]

SPEAKER                                           SENTENCE       DATAID  \
50      고객                          그러면 제일 빨리 나오는 게 어떤 짬뽕이에요?  박은영_전화_중식_7   
51      고객     그러면 4인 가족이 시켜 먹으려고 하는데 짬뽕 어떻게 시키나요 세트 메뉴도 있나요?  박은영_전화_중식_7   
52      고객                             중자에 다른 거는 뭐 안 시켜도 되나요?  박은영_전화_중식_7   
53      점원                          다른 음식 드시면서 탕수육 드실 거 아니에요?  박은영_전화_중식_7   
54      고객                            아 네 그러면 탕수육 돼지고기가 국산이죠?  박은영_전화_중식_7   
55      고객                                     아 그러면 면은 수타에요?  박은영_전화_중식_7   
56      고객                                    배달비는 따로 드려야 돼요?  박은영_전화_중식_7   
57      고객                                     선생님은 별도로 시키려구요  허정화_전화_피자_1   
58      고객                                   혹시 도우 두께라든지 치즈는?  허정화_전화_피자_1   
59      점원                                      네 혹시 어느 학교예요?  허정화_전화_피자_1   
60      고객                            할인 쿠폰이나 적립 카드 이런 건 있어요?  허정화_전화_피자_1   
61      고객                                  그러면 그거 10개 모으면은 또  허정화_전화_피자_1   
62      고객                                      배달료는 따로 받으세요?  허정화_전화_피자_1   
63      고객                                      종이컵도 따로 오는가요?  허정화_전화_피자_1   
64      고객                                   앱으로 주문도 할 수 있나요?  허정화_전화_피자_1   
65      고객                                   그럼 콜라 추가로는 얼마예요?  허정화_전화_피자_1   
66      점원                                          먼 곳는 아니죠?  허정화_전화_피자_1   
67      고객                                  슈퍼수프림은 위에 뭐 들어가요?  허정화_전화_피자_2   
68      고객                                     도우 두께나 그런 거는요?  허정화_전화_피자_2   
69      고객                                     불고기피자는 뭐 들어가요?  허정화_전화_피자_2   
70      고객                                 불고기는 그러면 국내산은 아니죠?  허정화_전화_피자_2   
71      고객                                 가공한 거네요 고기 자체가 아니고  허정화_전화_피자_2   
72      고객                                 블랙앵거스는 어떤 거 들어가지요?  허정화_전화_피자_2   
73      고객                                그거는 도우 두께 정할 수 있나요?  허정화_전화_피자_2   
74      고객                             피자 옆에 치즈 테두리에 들어가는 거는?  허정화_전화_피자_2   
75      고객                                         얼마나 더해지나요?  허정화_전화_피자_2   
76      고객                                   배달하면 배달료 따로 붙나요?  허정화_전화_피자_2   
77      고객                                     그럼 할인은 혹시 됩니까?  허정화_전화_피자_2   
78      고객                           만약에 통신사 카드 없이는 할인 안되는가요?  허정화_전화_피자_2   
79      고객                                        가서 가지러 가면요?  허정화_전화_피자_2   
80      고객                                        페트병은 얼마인가요?  허정화_전화_피자_2   
81      고객                            앱으로 배달하면 할인되고 그런 거 있어요?  허정화_전화_피자_2   
82      고객                       전화로 하는 거나 앱으로 하는 거나 똑같은 거예요?  허정화_전화_피자_2   
83      고객                             그러면 찾으러 가는게 할인 많이 되네요?  허정화_전화_피자_2   
84      고객                                     아침 몇 시에 문 열어요?  김옥연_전화_치킨_1   
85      고객                                           토요일은 하세요  김옥연_전화_치킨_1   
86      고객                        프라이드 세 마리하고 양념 두 마리 하면 얼마죠?  김옥연_전화_치킨_1   
87      고객                                 배달하고 포장하고 가격은 달라요?  김옥연_전화_치킨_1   
88      고객                                    그러면 포장은 얼마인가요?   김옥연_전화_치킨_1   
89      고객                                포장 시 무를 추가하면 얼마인가요?  김옥연_전화_치킨_1   
90      고객                   그렇구나 배달에는 그러면 무하고 콜라하고 같이 들어가고요?  김옥연_전화_치킨_1   
91      고객       그러니까 12000원 프라이드 값에 소스가 소금 나가고 또 포장하면 뭐 있어요?  김옥연_전화_치킨_1   
92      고객                               다른 거는 추가로 더 사야 한다고요?  김옥연_전화_치킨_1   
93      고객                                              콜라는요?  김옥연_전화_치킨_1   
94      고객                                      그러면 포장에는 따로네요  김옥연_전화_치킨_1   
95      고객  네 저녁에 손님 오면 시키려고 하는데요. 후라이드 종류는 몇 가지에요? 부위 가요....  김옥연_전화_치킨_2   
96      고객                                    부위별로 하는 거는 없어요?  김옥연_전화_치킨_2   
97      고객                              그러면 양념 된 거는 부위별로 있고요?  김옥연_전화_치킨_2   
98      고객                                  네 아 날개만 된 게 윙인가요?  김옥연_전화_치킨_2   
99      고객                                       날개하고 봉은 뭐예요?  김옥연_전화_치킨_2   

   DOMAINID DOMAIN CATEG

In [66]:
df_1= pd.DataFrame({'sentence':df_combine['SENTENCE']
                     , 'main':df_combine['MAIN']
                     })
df_1.head()

sentence         main
0                     지금 배달되나요?       배달가능문의
1  짬뽕류는 어떤 게 있나요? 잘 나가는 짬뽕 있나요?  베스트메뉴문의추천요청
2     전복 들어가는 거는 특해물 짬뽕 시켜야 돼요?       메뉴정보문의
3    전복 들어가고 여러 가지 또 딴 것도 들어가죠?       메뉴정보문의
4          마차이 짬뽕밥은 돼지고기 들어가나요?       메뉴정보문의

In [67]:
df_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34911 entries, 0 to 34910
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   sentence  34911 non-null  object
 1   main      34911 non-null  object
dtypes: object(2)
memory usage: 545.6+ KB


In [68]:
#'의도' 컬럼 따로 생성
df_1.loc[df_1[df_1.main.str.contains('문의')].index, 'meaning'] = '문의'
df_1.loc[df_1[df_1.main.str.contains('추천')].index, 'meaning'] = '추천'

In [69]:
df_1[250:300]

sentence         main meaning
250                                          가격은 얼마에요?         가격문의      문의
251                           아 모자라요 그러면 그 콜라는 추가로 오나요        서비스문의      문의
252                                        세트로 시켜야 되나요    세트메뉴서비스문의      문의
253                                 세트 메뉴는 어떤 게 잘 나가요?  베스트메뉴문의추천요청      추천
254                           원대동인데 주문할 때 배달비 드려야 되나요?        배달비문의      문의
255                                  주문하면 시간은 얼마쯤 걸려요?       배달시간문의      문의
256                                          피자 배달되나요?       배달가능문의      문의
257                                        한 판에 얼마인가요?         가격문의      문의
258                                치즈 크러스트로 말씀하시는 거예요?         가격문의      문의
259                                    네 종류가 어떤 거 있나요?         메뉴문의      문의
260                                 토핑은 위에 어떤 거 얹어지나요?         토핑문의      문의
261                               아 골라요 어떤 거 어떤 거 있어요?         토핑문의      문의
262                                   잘 나가는거 어떤 거 있어요?  베스트메뉴문의추천요청      추천
263                        네 그러면 주문하면 침산동인데 배달은 가능한가요?     배달장소범위문의      문의
264                                        배달비 따로 드려요?        배달비문의      문의
265                                 저 혹시 오픈시간이 어떻게 돼예?       오픈시간문의      문의
266                                      아 그럼 배달도 되구예?       배달가능문의      문의
267                                 그럼 배달료 따로 받던데 받나요?        배달비문의      문의
268                                   아 그럼 위치가 어떻게 돼요?       매장위치문의      문의
269                                         아 상가에 있어요?       매장위치문의      문의
270                               아 그러면 열두 시부터 배달은 돼요?       배달가능문의      문의
271                         아 그럼 순한 맛으로 하면 가격이 어떻게 돼요?         가격문의      문의
272                                          숯불로 굽습니까?      만드는방법문의      문의
273                  거기 비비큐 오리지널 후라이드랑 황금올리브유 차이가 뭐에요?       메뉴차이문의      문의
274                             그럼 거기 낮에 몇 시에 오픈해요 내일?       오픈시간문의      문의
275                          아 혹시 배달료 따로 받던데 배달료는 있나요?        배달비문의      문의
276                                    그럼 위치가 어디쯤 되나요?       매장위치문의      문의
277                                           뭐뭐 있는데예?         메뉴문의      문의
278                                         몇시에 오픈하세요?       오픈시간문의      문의
279                                     혹시 배달료 따로 받나요?        배달비문의      문의
280                                       배달료 얼마 받는데요?        배달비문의      문의
281                                   아 간장치킨 가격은 얼만데요?         가격문의      문의
282                             그 불갈비 치킨은 시간이 얼마나 걸려예?       배달시간문의      문의
283                                   배달하면 배달료 따로 있나요?        배달비문의      문의
284                                       거기 위치가 어디라예?       매장위치문의      문의
285                                       오픈은 몇 시에 해요?       오픈시간문의      문의
286                                      아 그럼 어디까지 해요?     배달장소범위문의      문의
287                                  그럼 퀵은 배송비 얼마 들어요?        배달비문의      문의
288                    그럼 통닭 값이랑 거기다가 오천 원 더 들어간다는 거죠?        배달비문의      문의
289                 그 후라이드 시킬 건데 요기요 통하면 50프로 디씨된다 하던데       메뉴할인문의      문의
290  네 제가 혹시 요기요로 주문했는데 배달할인 7000원이 되나요? 어제까지는 반반인데...       메뉴할인문의      문의
291    그러면 황금 올리브 치킨이나 아니면 그냥 반반 시킬려고 하는데 어떤 게 더 잘나가요?  베스트메뉴문의추천요청      추천
292                                       그 슈프림 치킨 있죠?         메뉴문의      문의
293                                    어떤 게 잘 나가요? 요즘에  베스트메뉴문의추천요청      추천
294                                       뭐가 들어가죠? 슈프림        식재료문의      문의
295                          후라이드가 더 잘나가요? 양념이 더 잘나가요?  베스트메뉴문의추천요청      추천
296                     아 그래요 후라이드 플러스 슈프림치킨 반반하면 얼마죠?         가격문의      문의
297                                      요기 배달료 포함인가요?        배달비문의      문의
298                                           두 개 하면은?         메뉴문의      문의
299                       제가 지금 결제해놓으면 이따가 배달해주실 수 있죠?       예약배달문의      문의

In [70]:
df_1['meaning'].value_counts()

문의    33176
추천     1735
Name: meaning, dtype: int64

In [71]:
df_1[df_1.meaning =='추천'].main.unique()

array(['베스트메뉴문의추천요청', '제품추천문의', '코디제품추천문의', '제품별추천문의', '유사제품추천문의',
       '연령에따른제품추천문의', '용도에따른제품추천문의', '추천상품 문의', '추천 펌 문의', '추천 커트 문의',
       '원하는 펌기술이나 기구가 없을 경우 가능한 미용실 추천 문의', '모발관리 추천 서비스 문의',
       '탈모관련 추천 서비스 문의', '탈모관련 추천 케어 서비스 문의', '취향에따른메뉴추천요구', '메뉴추천요청',
       '용도에따른제품추천요청', '연령에따른제품추천요청', '제품추천요청', '유사제품추천요청', '디자인별제품추천요청',
       '특정브랜드스타일추천요청', '선물용추천요청', '구매제품과코디할의류제품추천요청', '보조제품추천요청',
       '프로그램추천요청', '교육과정추천', '어학연수국가추천요청', '취급하는 제품 추천 요청', '제품추천 요청',
       '효과별 제품 추천', '반찬 추천 요청', '소비자 특성에 따른 추천 요청', '빵 추천 요청',
       '선물용 떡 추천 요청', '연령별 떡 추천 요청', '떡 추천 요청', '고기요리 추천 요청', '케이크 추천 요청',
       '선물용 추천', '저렴한 제품 추천 요청', '조합 제품 추천', '현재 본인 스타일의 염색 색상 추천 요청',
       '의약품 추천 요청', '선생님 추천 요청', '어울리는 헤어스타일 추천 요청', '헤어 길이별 스타일 추천 요청',
       '모발관리 추천 제품 요청', '의약품 추천 요정', '수선집 추천 요청', '메뉴추천요구'], dtype=object)

In [72]:
df_1[df_1.meaning =='문의'].main.unique()

array(['배달가능문의', '메뉴정보문의', '배달장소범위문의', ..., '쿠폰 사용 문의', '기본기 문의',
       '수강료 문의'], dtype=object)

In [73]:
len(df_1[df_1.meaning =='문의'].main.unique())

1690

In [74]:
df_1.to_csv('question_tag.csv', encoding='utf-8-sig')

In [75]:
question_tag = pd.read_csv('./question_tag.csv', index_col=0, encoding='utf-8-sig')

In [76]:
question_tag.tail()

sentence    main meaning
34906           과일 들어간 음료 하나 추천해 주세요  메뉴추천요구      추천
34907             가장 잘 나가는 음료 추천해주세요  메뉴추천요구      추천
34908         시원한 음료 중 뭐가 제일 잘 나가나요?  메뉴추천요구      추천
34909           많이 달지 않은 케익 좀 추천해주세요  메뉴추천요구      추천
34910  풍미가 좋은 커피 원두로 만든 커피 추천 가능할까요?  메뉴추천요구      추천

In [77]:
question_tag.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 34911 entries, 0 to 34910
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   sentence  34911 non-null  object
 1   main      34911 non-null  object
 2   meaning   34911 non-null  object
dtypes: object(3)
memory usage: 1.1+ MB


In [78]:
korean_re = re.compile('[^ ㄱ-ㅣ가-힣]+')

In [79]:
def extract_last_word(text):
    text = korean_re.sub('',text)
    text = text.split()[-1]
    return text

In [80]:
last_words = question_tag.sentence.apply(extract_last_word).tolist()
print(last_words)

['배달되나요', '있나요', '돼요', '들어가죠', '들어가나요', '배달되나요', '가능한가요', '나가요', '얼마에요', '돼지고기죠', '쓰나요', '나아요', '들어가요', '해요', '도착할까요', '가능한가요', '있어요', '들어가나요', '국산이에요', '거예요', '주시겠어요', '얼마에요', '되나요', '배달되나요', '나가나요', '하시는데요', '하는데', '가능한가요', '있어요', '사람들이', '따로', '얼마에요', '돼지고기죠', '쓰나요', '돼요', '해요', '닫아요', '배달되나요', '배달하시는데요', '있나요', '얼마에요', '시까지', '되나요', '해요', '들어가요', '세트메뉴', '돼요', '가능한가요', '있어요', '있어요', '짬뽕이에요', '있나요', '되나요', '아니에요', '국산이죠', '수타에요', '돼요', '시키려구요', '치즈는', '학교예요', '있어요', '또', '받으세요', '오는가요', '있나요', '얼마예요', '아니죠', '들어가요', '거는요', '들어가요', '아니죠', '아니고', '들어가지요', '있나요', '거는', '더해지나요', '붙나요', '됩니까', '안되는가요', '가면요', '얼마인가요', '있어요', '거예요', '되네요', '열어요', '하세요', '얼마죠', '달라요', '얼마인가요', '얼마인가요', '들어가고요', '있어요', '한다고요', '콜라는요', '따로네요', '하나요', '없어요', '있고요', '윙인가요', '뭐예요', '있나요', '얼마지요', '비싸요', '나오죠', '건가요', '오지요', '아니에요', '아닌가요', '되나요', '하려면요', '필요하신데요', '되나요', '가지죠', '먹었었는데', '있더라', '원이었던가', '올랐어요', '먹어봤는데', '바람에', '열어요', '하나요', '하세요', '있죠', '거', '있어요', '주시나요', '할인되죠', '걸려요', '얼마에요', '프라이드에', '할인되나요'

In [81]:
len(last_words)

34911

In [82]:
clean_last_words=list(set(last_words))

In [83]:
pprint(clean_last_words)

['준비되어있나요',
 '안나오나요',
 '올랐어요',
 '솜이에요',
 '판매하세요',
 '한가요',
 '소가죽',
 '강한데요',
 '않는건가요',
 '챠콜',
 '개드려요',
 '금지인가요',
 '했는데요',
 '수박있나요',
 '아몬드인가요',
 '전부인가요',
 '복잡하겠죠',
 '가르칩니까',
 '여쭈어볼려고요',
 '보시는데요',
 '해주시죠',
 '오신건가요',
 '계절음료인가요',
 '정도로',
 '할인혜택있나요',
 '타입이신가요',
 '묵을거에요',
 '하셨죠',
 '떨어졌네요',
 '당하셨나요',
 '있으셨어요',
 '달겠네요',
 '비슷해요',
 '어떤거요',
 '나가죠',
 '된다는거죠',
 '가시죠',
 '중에는',
 '여중인가요',
 '무겁네요',
 '드네요',
 '얼마정도해요',
 '아프셨나요',
 '받으셨나요',
 '사야겠죠',
 '원색상이에요',
 '셀프인가요',
 '거겠지요',
 '복합결제되나요',
 '사세요',
 '복숭아에요',
 '반찬이신가요',
 '카드되죠',
 '쓰셨나요',
 '쉬고',
 '독채인가요',
 '싶어가',
 '써보시려고요',
 '맞아요',
 '그러시나요',
 '분리되어있죠',
 '쌀은여기는다국산써요',
 '평일은',
 '싶어갖고요',
 '안되겠죠',
 '끊으면은',
 '수선하시나요',
 '그렇구나',
 '번',
 '하던데',
 '얼마에요',
 '그러거든요',
 '프린트되나요',
 '있는가',
 '내세요',
 '교재에요',
 '몇시까지에요',
 '수타면이에요',
 '자리마다는요',
 '만들었어요',
 '녹겠죠',
 '등록하실려고요',
 '생길까요',
 '현금가예요',
 '몇도입니까',
 '싱싱하겠네요',
 '아는가요',
 '올꼬',
 '시켜보려는데요',
 '단색이에요',
 '되어있으시고',
 '지방이겠죠',
 '세일하나요',
 '가입할게요',
 '있어여',
 '괜찮고요',
 '여자아이인가요',
 '어디신데요',
 '관리해요',
 '들어와요',
 '같던데',
 '약인가요',
 '맞지요',
 '배워야하죠',
 

 '제조되나요',
 '부작용은',
 '아포가토는',
 '올까요',
 '노세요',
 '앉을까예',
 '셀프구나',
 '여나요',
 '얼마고',
 '안만드시죠',
 '틀어주세요',
 '부탁드려요',
 '남아있나요',
 '해주실래요',
 '그건가요',
 '되셨는데요',
 '하려고요',
 '어디산이에요',
 '얇나요',
 '성인용인가요',
 '꺼예요',
 '뭐였어요',
 '분이라면서요',
 '해가지고요',
 '거에요',
 '톤이네',
 '만원씩인가요',
 '되는데예',
 '며칠간',
 '뽑아요',
 '시키시나요',
 '학생이세요',
 '송편저거는얼마예요',
 '지나가나요',
 '다',
 '계산해야하죠',
 '할인되는가요',
 '괜찮을가요',
 '했었어요',
 '살쪄보이나요',
 '상했나요',
 '얼마들어요',
 '내야하죠',
 '리터예요',
 '봐야하나요',
 '샴푸인가요',
 '싶어요',
 '안되는가요',
 '만원인거에요',
 '카피더라',
 '칠게요',
 '되도돼요',
 '시부터',
 '적립됐나요',
 '어딘데요',
 '케이크주문도돼요',
 '시부터에요',
 '사용한다고',
 '없던데요',
 '좋아하는데요',
 '되하는데얼마예요',
 '용도예요',
 '뭐에요',
 '걸려예',
 '아메리치노인가요',
 '어디인데요',
 '측정해드릴게요',
 '도넛인가요',
 '수업하는데요',
 '일분인가요',
 '할라고',
 '잘나가요',
 '몇일분이에요',
 '오셨나요',
 '써야해요',
 '할인행사해요',
 '유기농인가요',
 '샷이에요',
 '저거에요',
 '요즘좀많이해가는떡이어떤떡이에요',
 '추가되잖아요',
 '오신거죠',
 '할인은요',
 '삶으세요',
 '시작하죠',
 '갈라지거든요',
 '곳이에요',
 '병행한다죠',
 '되시는지예',
 '신상이죠',
 '겨울제품은',
 '여선생님인가요',
 '옮기셨어요',
 '안열리는데요',
 '어떠신지요',
 '피부관리하고',
 '오네요',
 '삼십만원요',
 '원이네요',
 '운동화에요',
 '펌하면',
 '사러왔습니다',
 '

In [84]:
len(clean_last_words)

5735